### PR TITLE
OpenGL threading refactor

### PR DESCRIPTION
**Pull request by hrydgard**
This is essentially the same as #16090 which did this for Vulkan, but this time it's for OpenGL.

I think the wins here are likely to be smaller but there's a clear win in making the code make more robust and make it easier to maintain. Got rid of a lot of weird state bools and stuff.

~~This needs some testing on Android before merge.~~

### CODE REVIEW

**Comment by hrydgard** [comment]
Not really sure why but this is currently very broken on some devices. Needs more work.

File: Common/GPU/OpenGL/GLRenderManager.cpp [hunk 1]
```diff
@@ -466,59 +412,55 @@ void GLRenderManager::CopyImageToMemorySync(GLRTexture *texture, int mipLevel, i
 void GLRenderManager::BeginFrame() {
 	VLOG("BeginFrame");
 
+	// TODO: Why is this debug-only?
```
Comment (unknownbrackets) [inline Common/GPU/OpenGL/GLRenderManager.cpp:421]: https://github.com/hrydgard/ppsspp/commit/42f4d7b40f0c86f996c764be0386d179dbb2aca3

-[Unknown]

File: Core/CoreTiming.cpp [hunk 2]
```diff
@@ -252,7 +250,12 @@ void Shutdown()
 
 u64 GetTicks()
 {
-	return (u64)globalTimer + slicelength - currentMIPS->downcount;
+	if (currentMIPS) {
+		return (u64)globalTimer + slicelength - currentMIPS->downcount;
```
Comment (unknownbrackets) [inline Core/CoreTiming.cpp:254]: I'll just fix this in the Reporting func.

-[Unknown]

File: Common/GPU/OpenGL/GLRenderManager.cpp [hunk 3]
```diff
@@ -181,133 +181,85 @@ void GLRenderManager::ThreadEnd() {
...
+		vrRenderStarted_ = false;
 	}
 
-	return true;
+	return didAnything;
 }
 
 void GLRenderManager::StopThread() {
-	// Since we don't control the thread directly, this will only pause the thread.
+	// There's not really a lot to do here anymore.
```
Comment (unknownbrackets) [inline Common/GPU/OpenGL/GLRenderManager.cpp:259]: Doesn't this at least need to add an EXIT?

-[Unknown]

File: Common/GPU/OpenGL/GLRenderManager.cpp [hunk 4]
```diff
@@ -181,133 +181,85 @@ void GLRenderManager::ThreadEnd() {
...
+
+		// We got a task! We can now have pushMutex_ unlocked, allowing the host to
+		// push more work when it feels like it, and just start working.
+		if (task.runType == GLRRunType::EXIT) {
+			// Oh, host wanted out. Let's leave, and also let's notify the host.
+			// This is unlike Vulkan too.
+			std::unique_lock<std::mutex> lock(syncMutex_);
+			syncCondVar_.notify_one();
+			syncDone_ = true;
+			break;
```
Comment (unknownbrackets) [inline Common/GPU/OpenGL/GLRenderManager.cpp:228]: Should this only break on exit, or should it always break on a frame (just not a sync)?

-[Unknown]

Comment (hrydgard) [reply Common/GPU/OpenGL/GLRenderManager.cpp:228]: Yeah, that turns out to be the main problem I was having - forgot that unlike the Vulkan function we really do have to bail here as soon as we want to do a present/SwapBuffers, as we don't have a SwapBuffers function to register.

**Comment by hrydgard** [comment]
SDL builds seem broken unforrtunately, needs more work.

**Comment by unknownbrackets** [comment]
Ideally should make sure headless (`--graphics=gles`) and CPU debugger breakpoints work too, as those were things that broke with Vulkan (I have not checked for this pull.)

-[Unknown]

**Comment by hrydgard** [comment]
Yup yup, not gonna get this in until those things are tested.

**Comment by hrydgard** [comment]
Don't understand what's going on here, the PPSSPP window stopped showing up on Mac. According to logging, if I turn it on, it all seems OK though...

**Comment by hrydgard** [comment]
Rebased on master. Still perfectly functional on PC and Android - just still broken on Mac. Very strange.

**Comment by hrydgard** [comment]
Alright! It works now, on all the platforms. Just a silly oversight.

**Pull request merged**